src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -809,31 +807,55 @@ private static LiftingResult UserDefinedBinaryOperatorCanBeLifted(TypeSymbol lef
+            int currentBestIndex;
             var candidates = result.Results;
-            for (int i = 1; i < candidates.Count; ++i)
+            for (currentBestIndex = 0; currentBestIndex < candidates.Count; currentBestIndex++)
+            {
+                if (candidates[currentBestIndex].Kind == OperatorAnalysisResultKind.Applicable)
+                {
+                    break;
+                }
+            }
+
+            for (int index = currentBestIndex + 1; index < candidates.Count; index++)
             {
-                if (candidates[i].Kind != OperatorAnalysisResultKind.Applicable)
+                if (candidates[index].Kind == OperatorAnalysisResultKind.Inapplicable)
                 {
                     continue;
                 }
 
-                // Is this applicable operator better than every other applicable method?
-                for (int j = 0; j < i; ++j)
+                var better = BetterOperator(candidates[currentBestIndex].Signature, candidates[index].Signature, left, right, ref useSiteDiagnostics);
+                if (better == BetterResult.Left)
                 {
-                    if (candidates[j].Kind == OperatorAnalysisResultKind.Inapplicable)
-                    {
-                        continue;
-                    }
-                    var better = BetterOperator(candidates[i].Signature, candidates[j].Signature, left, right, ref useSiteDiagnostics);
-                    if (better == BetterResult.Left)
-                    {
-                        candidates[j] = candidates[j].Worse();
-                    }
-                    else if (better == BetterResult.Right)
-                    {
-                        candidates[i] = candidates[i].Worse();
-                    }
+                    candidates[index] = candidates[index].Worse();
+                }
+                else if (better == BetterResult.Right)
+                {
+                    candidates[currentBestIndex] = candidates[currentBestIndex].Worse();
+                    currentBestIndex = index;
+                }
+            }
+
+            for (int index = 0; index < candidates.Count; index++)
+            {
+                if (index == currentBestIndex || candidates[index].Kind == OperatorAnalysisResultKind.Inapplicable)
+                {
+                    continue;
+                }
+
+                var better = BetterOperator(candidates[currentBestIndex].Signature, candidates[index].Signature, left, right, ref useSiteDiagnostics);
+                if (better == BetterResult.Left)
+                {
+                    candidates[index] = candidates[index].Worse();
+                }
+                else if (better == BetterResult.Right)
+                {
+                    candidates[currentBestIndex] = candidates[currentBestIndex].Worse();
+                    break;
+                }
+                else
+                {
+                    break;
```

Comment:
Obviously no. Are you suggesting that the algorithm is incomplete? Not if betterness is transitive (which is probably not, see CircularImplicitConversions test added in this PR). If you confirm that, I'll review my implementation accordingly later today.